### PR TITLE
Add the RPC to execute transactions

### PR DESCRIPTION
**Pull request by sgkim126**

### CODE REVIEW

**Comment by sgkim126** [comment]
Depends on https://github.com/CodeChain-io/codechain/pull/446

**Comment by sgkim126** [comment]
I forgot to describe it on spec. I'll update the PR.

File: spec/JSON-RPC.md [hunk 1]
```diff
@@ -691,6 +697,41 @@ Response Example
...
+Params:
+1. transactions: `TransactionObject[]`
+
+Return Type: `ChangeShardObject[]`
+
+Request Example
+```
+  curl \
+    -H 'Content-Type: application/json' \
+    -d '{"jsonrpc": "2.0", "method": "chain_executeTransactions", "params": ["0xf8c8f8630311809e6d65746164617461206f66207065726d697373696f6e6564206173736574a007feab4c39250abf60b77d7589a5b61fdf409bd837e936376381d19db1e1f050c0c7865af3107a4000d5943f4aa1fedf1f54eeb03b759deadb36676b18491180f861031101a26d65746164617461206f66206e6f6e2d7065726d697373696f6e6564206173736574a007feab4c39250abf60b77d7589a5b61fdf409bd837e936376381d19db1e1f050c0c164d5943f4aa1fedf1f54eeb03b759deadb36676b18491180"], "id": null}' \
```
Comment (joojis) [inline spec/JSON-RPC.md]: Is it the RLP encoding of transactions?

File: core/src/client/client.rs [hunk 2]
```diff
@@ -274,6 +276,22 @@ impl AssetClient for Client {
...
 }
 
+impl ExecuteClient for Client {
+    fn execute_transactions(&self, transactions: &[Transaction]) -> Result<Vec<ChangeShard>, Error> {
+        let state = Client::state_at(&self, BlockId::Latest).expect("Latest state MUST exist");
+        let mut shard_ids: Vec<u32> = transactions.iter().flat_map(Transaction::related_shards).collect();
+        shard_ids.sort_unstable();
+        shard_ids.dedup();
+
+        let mut result = Vec::default();
```
Comment (kseo) [inline core/src/client/client.rs]: Can't we use `map` and `collect` here?

File: rpc/src/v1/errors.rs [hunk 3]
```diff
@@ -70,6 +70,15 @@ pub fn parcel_core<T: Into<CoreError>>(error: T) -> Error {
     }
 }
 
+pub fn core<T: Into<CoreError>>(error: T) -> Error {
+    let error = error.into();
+    Error {
+        code: ErrorCode::ServerError(codes::KVDB_ERROR),
```
Comment (kseo) [inline rpc/src/v1/errors.rs]: Why `KVDB_ERROR`?

**Comment by sgkim126** [comment]
@kseo pr updated

**Pull request merged**